pkg/issuer/acme/dns/dns.go

```diff
@@ -292,6 +292,30 @@ func (s *Solver) solverForChallenge(ctx context.Context, issuer v1.GenericIssuer
 		}
 	case providerConfig.Route53 != nil:
 		dbg.Info("preparing to create Route53 provider")
+
+		// For route53, you must specify either an AccessKeyID or a secret reference to an AccessKeyID
+		// but not both.
+		if providerConfig.Route53.AccessKeyID != "" && providerConfig.Route53.SecretAccessKeyID.Name != "" {
```

Comment:
Whilst we should also have this logic here, it _also_ needs to exist in the validation webhook/as part of our API - this will mean that if a user attempts to submit a resource that _does_ set both, it won't be admitted into the apiserver in the first place.

These changes should be made in the API validation package here: https://github.com/cert-manager/cert-manager/blob/c16b3cca7b418ba0d0b2bf1066514b8762984517/internal/apis/certmanager/validation/issuer.go#L401-L407

and you can add some test cases for this code here: https://github.com/cert-manager/cert-manager/blob/c16b3cca7b418ba0d0b2bf1066514b8762984517/internal/apis/certmanager/validation/issuer_test.go#L708-L715